generate.py

```diff
@@ -57,9 +59,13 @@ def generate(
 
         probs = torch.nn.functional.softmax(logits, dim=-1)
         idx_next = torch.multinomial(probs, num_samples=1)
-
+        
+        # if <eos> token is triggered, return the output (stop generation)
+        if idx_next==eos_id:
```

Comment:
This is batched so this wont work. See my comment at https://github.com/Lightning-AI/lit-llama/issues/130#issuecomment-1505564783

Reply:
Ah, of course you’re right.
this speeds up generation, so I’d vouch for following this approach